ui/src/chatbot.test.tsx

```diff
@@ -159,4 +159,72 @@ describe('XChatbot', () => {
     expect(emitMock).toHaveBeenCalledTimes(1)
     expect(emitMock).toHaveBeenCalledWith(model.name, 'scroll_up', true)
   })
+
+  it('Renders thumbs up/down buttons', () => {
+    const { container } = render(<XChatbot {...{ ...model, data, events: ['feedback'] }} />)
+    const likeButton = container.querySelector("i[data-icon-name='Like']") as HTMLLIElement
+    const dislikeButton = container.querySelector("i[data-icon-name='Dislike']") as HTMLLIElement
+    expect(likeButton).toBeInTheDocument()
+    expect(dislikeButton).toBeInTheDocument()
+  })
+
+  it('Fires a like feedback event when clicked on the thumbs up button', () => {
+    const { container } = render(<XChatbot {...{ ...model, data, events: ['feedback'] }} />)
+    const likeButton = container.querySelector("i[data-icon-name='Like']") as HTMLLIElement
+
+    fireEvent.click(likeButton)
+    expect(emitMock).toHaveBeenCalled()
+    expect(emitMock).toHaveBeenCalledTimes(1)
+    expect(emitMock).toHaveBeenCalledWith(model.name, 'feedback', { message: data[1].content, positive: true })
+
+    const likeSolidButton = container.querySelector("i[data-icon-name='LikeSolid']") as HTMLLIElement
+    expect(likeSolidButton).toBeInTheDocument()
+  })
+
+  it('Fires a dislike feedback event when clicked on the thumbs down button', () => {
+    const { container } = render(<XChatbot {...{ ...model, data, events: ['feedback'] }} />)
+    const dislikeButton = container.querySelector("i[data-icon-name='Dislike']") as HTMLLIElement
+
+    fireEvent.click(dislikeButton)
+    expect(emitMock).toHaveBeenCalled()
+    expect(emitMock).toHaveBeenCalledTimes(1)
+    expect(emitMock).toHaveBeenCalledWith(model.name, 'feedback', { message: data[1].content, positive: false })
+
+    const dislikeSolidButton = container.querySelector("i[data-icon-name='DislikeSolid']") as HTMLLIElement
+    expect(dislikeSolidButton).toBeInTheDocument()
+  })
+
+  it('Fires a no feedback event when clicked on the thumbs up button twice', () => {
```

Comment:
Like/Dislike should be fired at most once unless like => dislike => like etc.

Reply:
So undoing of like/dislike shouldn't be possible?

Reply:
Correct.